src/adapters/rubic/execution/src/index.ts

```diff
@@ -0,0 +1,129 @@
+import { RubicSwappedGenericSchema, RubicSwappedGenericsRes, RubicTransferStartedSchema, RubicTransferStartedsRes, UserTxData } from "./types"
+import { JsonRpcProvider, ethers } from 'ethers'
+import ERC20_ABI from "./ERC20.json"
+
+const RUBIC_MULTI_PROXY_ADDRESS = '0x1a979E2386595837BaAB90Ba12B2E2a71C652576'
+const ZK_LINK_CHAIN_ID = 810180;
+
+export const getUserTransactionData = async (startBlock: number, endBlock: number): Promise<UserTxData[]> => {
+  console.log(`Get Tx Data From ${startBlock} to ${endBlock}`)
+
+  const manager = new RubicQueriesManager(startBlock, endBlock)
+  const [onchainSwaps, ccrSwaps] = await Promise.all([
+      manager.queryRubicSwappedGenerics(),
+      manager.queryRubicTransferStarteds()
+  ])
+  const transactions = [...onchainSwaps, ...ccrSwaps]
+
+  return transactions
+};
+
+class RubicQueriesManager {
+  private _provider = new JsonRpcProvider('https://rpc.zklink.io')
+
+  private _startBlock: number
+
+  private _endBlock: number
+
+  constructor(start: number, end: number) {
+    this._startBlock = start;
+    this._endBlock = end;
+  }
+
+  public async queryRubicSwappedGenerics(): Promise<UserTxData[]> {
+    const query = `query MyQuery {
+      rubicSwappedGenerics(where: {
+        blockNumber_gte: ${this._startBlock},
+        blockNumber_lte: ${this._endBlock}
+      }){
+        fromAssetId
+        fromAmount
+        transactionHash
+        blockTimestamp
+        blockNumber
+      }
+    }
+    `
+    const response = await fetch(
+      "http://3.114.68.110:8000/subgraphs/name/rubic-finance-2", 
+    {
+      method: 'POST',
+      body: JSON.stringify({ query }),
+      headers: { 'Content-Type': 'application/json' }
+    }) as unknown as RubicSwappedGenericsRes;
+
+    const transfers = response.data?.rubicSwappedGenerics || []
+    const promises = transfers.map(el => this.mapRSG(el))
+    const res = await Promise.all(promises)
+    
+    return res
+  }
+
+  public async queryRubicTransferStarteds(): Promise<UserTxData[]> {
+    const query = `query MyQuery {
+      rubicTransferStarteds(where: {
+        blockNumber_gte: ${this._startBlock},
+        blockNumber_lte: ${this._endBlock}
+      }) {
+        bridgeData_sendingAssetId
+        bridgeData_destinationChainId
+        bridgeData_receiver
+        bridgeData_minAmount
+        blockNumber
+        blockTimestamp
+        transactionHash
+    }
+    `
+    const response = await fetch(
+      "http://3.114.68.110:8000/subgraphs/name/rubic-finance-2", 
+    {
+      method: 'POST',
+      body: JSON.stringify({ query }),
+      headers: { 'Content-Type': 'application/json' }
+    }) as unknown as RubicTransferStartedsRes;
+
```

Comment:
The TypeScript assertion here caused you the issue. You should call `await data.json()`. 
Suggest to run `node runScript.js rubic 2225434 0` in the adapter directory. If it runs correctly, it will create `tx.2225434.csv` in the `rubic/data` directory with data inside. I ran the code from your submitted branch, and there is no data. Please check the execution code. 

Additionally, refer to the example to add retry functionality for subgraph queries and ensure the subgraph URL comes from the environment variables.

Reply:
done